core/create/src/prompts/oidc.ts

```diff
@@ -0,0 +1,274 @@
+        serviceName
+      )
+      if (previousDocument) {
+        winstonLogger.info(
+          'Found the permissions assigned to an IAM user for CI deployments previously. Reusing these permissions for your OIDC role policy.'
+        )
+        cloudformationTemplate.setIn(
+          ['Resources', 'CircleciOIDCRolePolicy', 'Properties', 'PolicyDocument'],
+          previousDocument
+        )
+      } else {
+        winstonLogger.info(
+          "Couldn't find any old IAM permissions to reuse for this project. Falling back to the default permissions recommended by Cloud Enablement."
+        )
+      }
+    }
+  }
+
+  let circleciProjectId
+  try {
+    circleciProjectId = await getCircleCiProjectId(vaultEnv.CIRCLECI_AUTH_TOKEN)
+  } catch (err) {
+    winstonLogger.error(`Failed to automatically determine the ID for your CircleCI prject: ${err}`)
+    circleciProjectId = (
+      await prompt(
+        {
+          name: 'circleId',
+          type: 'text',
+          message:
+            "You can instead enter the project ID manually. You can find the ID by going to your project's settings in the CircleCI UI and copying the ID displayed at Overview > Project ID."
+        },
+        { onCancel }
+      )
+    ).circleId
+    if (cancelled) {
+      return true
+    }
+  }
+
+  const templatePath = `cloudformation/${teamName}/${stackTags.systemCode}-deployment-role.yaml`
+  const configPath = `config/composer/custom/${teamName}/global/${stackTags.systemCode}-deployment-role.yaml`
+
+  const composerConfig = {
+    template_path: templatePath,
+    parameters: {
+      SystemCode: stackTags.systemCode,
+      CircleciProjectId: circleciProjectId,
+      CircleciOrgName: 'Financial-Times'
+    },
+    stack_tags: stackTags
+  }
+
+  const upstreamRepo = `aws-composer-account-${awsAccount}`
+  const changes = new Map([
+    [templatePath, { mode: FILE_MODE, content: cloudformationTemplate.toString() }],
+    [configPath, { mode: FILE_MODE, content: YAML.stringify(composerConfig) }]
+  ])
+  winstonLogger.info(
+    `Will send the following changes to the ${styles.URL(
+      `https://github.com/Financial-Times/${upstreamRepo}`
+    )} repository: ${changes}`
+  )
+  const { prConfirm } = await prompt(
+    { name: 'prConfirm', type: 'confirm', message: 'Is that okay?' },
+    { onCancel }
+  )
+  if (prConfirm) {
+    const octokit = new Octokit({ auth: vaultEnv.GITHUB_ACCESS_TOKEN })
+    await suggester.createPullRequest(octokit, changes, {
+      upstreamRepo,
+      upstreamOwner: 'Financial-Times',
+      fork: false,
+      title: `Add OIDC authentication for ${stackTags.systemCode}`,
+      description: `This PR has been automatically generated via Tool Kit. It adds the deployment role necessary for the ${stackTags.systemCode} project to deploy to AWS via OIDC authentication, as well as the associated aws-composer configuration.`,
+      message: `feat: add OIDC authentication for ${stackTags.systemCode}`
+    })
```

Comment:
**suggestion:** Can we mention the current GitHub user in the PR details? This will mean the generated PR is clearly linked to the person who created it.

Reply:
great idea!